messaging/messaging.go

```diff
@@ -398,6 +398,63 @@ func (a *Aps) MarshalJSON() ([]byte, error) {
 	return json.Marshal(m)
 }
 
+// UnmarshalJSON unmarshals an Aps from JSON (for internal use only).
+func (a *Aps) UnmarshalJSON(data []byte) error {
+	m := make(map[string]interface{})
+	a.CustomData = make(map[string]interface{})
+	err := json.Unmarshal(data, &m)
+	if err != nil {
+		return err
+	}
+	for k, v := range m {
+		switch k {
+		case "alert":
+			if v != nil {
+				switch alert := m["alert"].(type) {
+				case interface{}:
+					a.Alert = alert.(*ApsAlert)
```

Comment:
This will panic. Type of alert is either `string` or `map[string]interface{}` at this point.